python/dgl/nn/pytorch/explain/pgexplainer.py

```diff
@@ -0,0 +1,351 @@
+    sample_bias : float, optional
+        Some members of a population are systematically more likely to be selected
+        in a sample than others. Default: 0.0.
+    """
+
+    def __init__(
+        self,
+        model,
+        num_features,
+        coff_budget=0.01,
+        coff_connect=5e-4,
+        sample_bias=0.0,
+    ):
+        super(PGExplainer, self).__init__()
+
+        self.model = model
+        self.num_features = num_features * 2
+
+        # training hyperparameters for PGExplainer
+        self.coff_budget = coff_budget
+        self.coff_connect = coff_connect
+        self.sample_bias = sample_bias
+
+        self.init_bias = 0.0
+
+        # Explanation network in PGExplainer
+        self.elayers = nn.ModuleList()
+        self.elayers.append(
+            nn.Sequential(nn.Linear(self.num_features, 64), nn.ReLU())
+        )
+        self.elayers.append(nn.Linear(64, 1))
+
+    def set_masks(self, graph, feat, edge_mask=None):
+        r"""Set the edge mask that play a crucial role to explain the
+        prediction made by the GNN for a graph. Initialize learnable edge
+        mask if it is None.
+
+        Parameters
+        ----------
+        graph : DGLGraph
+            A homogeneous graph.
+        feat : Tensor
+            The input feature of shape :math:`(N, D)`. :math:`N` is the
+            number of nodes, and :math:`D` is the feature size.
+        edge_mask : Tensor, optional
+            Learned importance mask of the edges in the graph, which is a tensor
+            of shape :math:`(E)`, where :math:`E` is the number of edges in the
+            graph. The values are within range :math:`(0, 1)`. The higher,
+            the more important. Default: None.
+        """
+        num_nodes, _ = feat.shape
+        num_edges = graph.num_edges()
+
+        init_bias = self.init_bias
+        std = nn.init.calculate_gain("relu") * math.sqrt(2.0 / (2 * num_nodes))
+
+        if edge_mask is None:
+            self.edge_mask = torch.randn(num_edges) * std + init_bias
+        else:
+            self.edge_mask = edge_mask
+
+        self.edge_mask = self.edge_mask.to(graph.device)
+
+    def clear_masks(self):
+        r"""Clear the edge mask that play a crucial role to explain the
+        prediction made by the GNN for a graph.
+        """
+        self.edge_mask = None
+
+    def loss(self, prob, ori_pred):
+        r"""The loss function that is used to learn the edge
+        distribution.
+
+        Parameters
+        ----------
+        prob:  Tensor
+            Tensor contains a set of probabilities for each possible
+            class label of some model.
+        ori_pred: int
+            Integer representing the original prediction.
+
+        Returns
+        -------
+        float
+            The function returns the sum of the three loss components,
+            which is a scalar tensor representing the total loss.
+        """
+        logit = prob[ori_pred]
```

Comment:
how about "target_prob"?

the use of "logit" can be confusing as people may assume this to be pre-softmax

Reply:
fixed.